clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -318,4 +321,16 @@ public AttributeList setAttributes(AttributeList list) {
                                       + ".(whitelist/blacklist) is not a valid regular expression");
         }
     }
+
+    @Override
+    public void contextChange(MetricsContext metricsContext) {
+        if (!this.mbeans.isEmpty()) {
+            throw new IllegalStateException("JMX MetricsContext can only be updated before JMX metrics are created");
+        }
+        synchronized (LOCK) {
+            if (this.prefix.isEmpty()) {
```

Comment:
since we check that no mbeans have been created, it should now be safe to update it whether or not it was empty before.